Source/WebCore/platform/gstreamer/GStreamerCodecUtilities.cpp

```diff
@@ -66,6 +67,31 @@ std::pair<const char*, const char*> GStreamerCodecUtilities::parseH264ProfileAnd
     return { profile, level };
 }
 
+const char* GStreamerCodecUtilities::parseHEVCProfile(const String& codec)
+{
+    ensureDebugCategoryInitialized();
+
+    GST_DEBUG("Parsing HEVC codec string: %s", codec.ascii().data());
+    auto parameters = parseHEVCCodecParameters(codec);
+    if (!parameters) {
+        GST_WARNING("Invalid HEVC codec: %s", codec.ascii().data());
+        return nullptr;
+    }
+
+    uint8_t profileTierLevel[11] = { 0, };
+    memset(profileTierLevel, 0, 11);
+    profileTierLevel[0] = parameters->generalProfileIDC;
+
+    if (profileTierLevel[0] < 4)
+        return gst_codec_utils_h265_get_profile(profileTierLevel, sizeof(profileTierLevel));
+
+    auto& constraints = parameters->generalConstraintIndicatorFlags;
+    for (unsigned i = 5, j = 0; i < 10; i++, j++)
+        profileTierLevel[i] = constraints[j];
+
+    return gst_codec_utils_h265_get_profile(profileTierLevel, sizeof(profileTierLevel));
```

Comment:
Since you're returning the same thing just after the constain copy, I would put the constain copy inside an `if (profileTierLevel[0] >= 4)` and then return after it.